designs/NuGet-ClientCertificates.md

```diff
@@ -0,0 +1,200 @@
+    - Child item `<Add Key="StoreName" Value="[values]" />`. [Possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storename?view=netframework-4.8#fields). Optional. Equals `My` by default.
+    - Child item `<Add Key="FindType" Value="[values]" />`. [Possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509findtype?view=netframework-4.8#fields). Optional. Equals `FindByThumbprint` by default.
+    - Child item `<Add Key="FindValue" Value="value" />`. `value` - the search criteria as an object. Required.
+
+2. `fromFile` - certificate import from file (DER encoded x.509, Base-64 encoded x.509, PKCS)
+Can be configured with:
+    - Attribute `name`. Required. Unique setting identifier.
+    - Child item `<Add Key="Path" Value="value" />`. `value` - Absolute or relative path to certificate file. Required.
+    - Child item `<Add Key="Password" Value="value" />`. `value` - Plain or encrypted password string. Optional. Encrypted in same manner as PackageSourceCredential password.
+
+## Configuration example
+
+```xml
+<configuration>
+...
+    <clientCertificates>	
+        <fromStorage name="unique name 1">
+            <!-- Require at least one reference to source -->
+            <add key="TargetSource" value="<source name from packageSources section>" />
+            <!-- Optional. CurrentUser by default -->
+            <add key="StoreLocation" value="CurrentUser" />
+            <!-- Optional. My by default -->
+            <add key="StoreName" value="My" />
+            <!-- Optional. FindByThumbprint by default -->
+            <add key="FindType" value="FindByThumbprint" />
+            <!-- Required. -->
+            <add key="FindValue" value="4894671ae5aa84840cc1079e89e82d426bc24ec6" />
+        </fromStorage>
+        <fromFile name="unique name 2">
+            <!-- Require at least one reference to source -->
+            <add key="TargetSource" value="<source name from packageSources section>" />
+            <!-- Absolute or relative path to certificate file. -->
+            <add key="Path" value=".\certificate.pfx" />
+            <!-- Encrypted password -->
+            <add key="Password" value="..." />
+        </fromFile>
+    </clientCertificates>
+...
+</configuration>
+```
+
+## client-certificates command
+
+Gets, sets or lists client certificates to the NuGet configuration.
+
+Usage
+```
+nuget client-certificates <list|add|remove> [options]
+```
+
+if none of `list|add|remove` is specified, the command will default to `list`.
+
+### nuget client-certificates list [options]
+
+Lists all the client certificates in the configuration. This option will include all configured client certificates that match to specified options.
+
+- `-Check` one of `true|false` - Indicates that certificate existence must be checked. If the certificate is found, its fingerprint will be printed.
```

Comment:
- What is the behavior when the `-Check` is not present ? 
- Would that always imply `-Check false` ? 
- Can we imply that if the user only writes `-Check`, it will mean `-Check true` ?

Reply:
> What is the behavior when the -Check is not present ?
Would that always imply -Check false ?

Yes

> Can we imply that if the user only writes -Check, it will mean -Check true

Reasonable

Reply:
Specification updated